pkg/localstorev2/storer.go

```diff
@@ -370,6 +374,138 @@ func (p *putterSession) Done(addr swarm.Address) error { return p.done(addr) }
 
 func (p *putterSession) Cleanup() error { return p.cleanup() }
 
+func (db *DB) SubscribePush(ctx context.Context) (chunks chan swarm.Chunk, reset, stop func()) {
+	chunks = make(chan swarm.Chunk)
+	trigger := make(chan struct{}, 1)
+	resetC := make(chan struct{}, 1)
+
+	// send signal for the initial iteration
+	trigger <- struct{}{}
+
+	db.pushTriggersMu.Lock()
+	db.pushTriggers = append(db.pushTriggers, trigger)
+	db.pushTriggersMu.Unlock()
+
+	stopChan := make(chan struct{})
+	var stopChanOnce sync.Once
+
+	var sinceItem swarm.Chunk
+
+	reset = func() {
+		time.Sleep(1 * time.Second) // give some time when retrying
```

Comment:
why do we sleep?

Reply:
no sure, this functionality was ported from the old subscriber

Reply:
with the rewrite, we have a good opportunity to step back, reconsider things with hindsight. I would advocate for a redesign  between the upload - pusher interactions IF the old mechanism are not up to par.